client/src/utils/fuzzyFilter.ts

```diff
@@ -1,27 +1,45 @@
-import { FilterFn } from '@tanstack/react-table'
-import { rankItem } from '@tanstack/match-sorter-utils'
+import { FilterFn, Row } from '@tanstack/react-table'
+import { rankItem, RankingInfo } from '@tanstack/match-sorter-utils'
+import { isEmpty } from 'lodash'
 
-export const fuzzyFilter: FilterFn<any> = (row, columnId, value, addMeta) => {
+const substringMatch = (value: string, searchTerm: string): boolean => {
+  const normalizedValue = value.toLowerCase()
+  const normalizedSearch = searchTerm.toLowerCase()
+  return normalizedValue.includes(normalizedSearch)
+}
+
+export const fuzzyFilter: FilterFn<any> = (
+  row: Row<any>,
+  columnId: string,
+  value: unknown,
+  addMeta: (meta: { itemRank: RankingInfo }) => void
+): boolean => {
   try {
-    // Get the value for the 'projects' column
+    if (isEmpty(value)) return true
+
+    if (columnId === 'all') {
+      const name = row.getValue('user.name')
+      if (isEmpty(name)) return false
+      return substringMatch(String(name), String(value))
```

Comment:
```suggestion
     return !isEmpty(name) && substringMatch(String(name), String(value))
```

Avoid nested ifs for better readability